passExtensions.pass-audit: fix python3.12 compat

gador opened the pull request:
## Description of changes

Fix for python 3.12:
```
> Traceback (most recent call last):
>   File "/build/tmpc8ccl7e8.py", line 1, in <module>
>     from distutils.util import byte_compile
> ModuleNotFoundError: No module named 'distutils'
```
Upstream has no new releases since last year and some custom logic in its `Makefile`. Since it basically is just a bash script wrapped python library, I refactored it to use our python build system.

@Ma27 for review

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Ma27 commented on pkgs/tools/security/pass/extensions/audit/default.nix:
```diff
@@ -22,29 +35,42 @@ in stdenv.mkDerivation rec {
 
-  buildInputs = [ pythonEnv ];
-  nativeBuildInputs = [ makeWrapper ];
+  dependencies = [ pythonEnv ];
```
If you use the python build system here, it's probably better to just list all the project dependencies in here, right?

gador replied:
I did think about it. But this project also uses the bash script `audit.bash` which needs the whole `pythonEnv`. It seems easier to declare the `pythonEnv` and just use it (like  you had it before)

Ma27 reviewed (commented):
For the future, it would be nice if you could split format changes and actual changes into two commits. That'd make reviewing way easier.

gador committed:
passExtensions.pass-audit: fix python3.12 compat

Signed-off-by: Florian Brandes <florian.brandes@posteo.de>

Ma27 committed:
passExtensions.pass-audit: get rid of pythonEnv

It seems a little odd to first generate an interpreter pointing to an
environment and then building a package from it. This commit alters the
build process so that

* dependencies are listed explicitly as in any normal python build.
* a PYTHONPATH is then generated out of all dependencies and $out. This
  is injected into the package's entrypoint, the `audit.bash`.

Ma27 commented:
Proposed a different solution not relying on an extra env. wdyt?

gador commented:
Looks cleaner than before! LGTM 👍

gador commented on pkgs/tools/security/pass/extensions/audit/default.nix:
```diff
@@ -34,7 +26,7 @@ pythonPackages.buildPythonApplication rec {
   postPatch = ''
     substituteInPlace audit.bash \
-      --replace-fail 'python3' "${pythonEnv.interpreter}"
+      --replace python3 "${lib.getExe python3}"
```
this should be `replace-fail`

Ma27 replied:
eh, `--replace` is so deep in my muscle memory, it will take me a while to get rid of that ;-)

Thanks!
